vm/src/obj/objbool.rs

```diff
@@ -18,7 +18,12 @@ impl IntoPyObject for bool {
 
 impl TryFromObject for bool {
     fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<bool> {
-        boolval(vm, obj)
+        if objtype::isinstance(&obj, &vm.ctx.int_type()) {
+            Ok(get_value(&obj))
+        } else {
+            let actual_type = vm.to_pystr(&obj.class())?;
```

Comment:
You should be able to get `obj.class().name`, and avoid the possibility of an error